### PR TITLE
neovim: Add vim-addon-nix to show broken build

### PR DESCRIPTION
**Pull request by matthiasbeyer**
CC #154299

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
@teto I can, of course, also make this into a completely new test (without deoplete) to show that it fails, if you want me to!

**Comment by teto** [comment]
Looks like a bug with our pathogen implementation in pkgs/misc/vim-plugins/vim-utils.nix . I suspect few people use it if you are the first to trigger the issue. I actually would like to get rid of these alternate implementations since now vim has packages but feel free to submit a PR if you use pathogen and want to keep using it. 
It's possible this bug is different than your original one though. 

Try with
```
  test_nvim_with_remote_plugin = neovim.override {
    extraName = "-pathogen-remote";
    configure.packages.foo.start = with vimPlugins; [ deoplete-nvim ];
  };
```

**Comment by matthiasbeyer** [comment]
Now it works. I do not actually care what plugin manager I use, so switching away from pathogen is no issue for me - I just wonder why my setup used pathogen in the first place :laughing:

File: pkgs/applications/editors/neovim/tests.nix [hunk 1]
```diff
@@ -154,6 +154,16 @@ rec {
     configure.pathogen.pluginNames = with vimPlugins; [ deoplete-nvim ];
```
Comment (teto) [inline pkgs/applications/editors/neovim/tests.nix:154]:
```suggestion
    configure.pathogen.pluginNames = with vimPlugins; [ deoplete-nvim vim-addon-nix  ];
```

Comment (teto) [reply pkgs/applications/editors/neovim/tests.nix:154]: and rename to ` extraName = "-remote";`

File: pkgs/applications/editors/neovim/tests.nix [hunk 2]
```diff
@@ -154,6 +154,16 @@ rec {
...
+  test_nvim_with_remote_plugin_non_pathogen = neovim.override {
+    extraName = "-nonpathogen-remote";
+    configure.packages.foo.start = with vimPlugins; [ deoplete-nvim ];
+  };
+
+  test_nvim_with_nix_plugin_non_pathogen = neovim.override {
+    extraName = "-nix-nonpathogen-remote";
+    configure.packages.foo.start = with vimPlugins; [ vim-addon-nix ];
+  };
+
```
Comment (teto) [inline pkgs/applications/editors/neovim/tests.nix:166]: remove these

**Comment by teto** [comment]
to sum up just swap the pathogen implementation with the native one. I will submit a PR to remove pathogen and see if people react (or at least add a warning to switch to native packagement). plug and vundle have extra features compared to vim packages taht I dont think pathogen has. If true, it makes no sense to keep it.

**Comment by matthiasbeyer** [comment]
> just swap the pathogen implementation with the native one

you mean in my local installation? I tried several approaches for configuring my neovim, none worked (either build failed or build succeeded but neovim didn't find any plugins).

**Comment by teto** [comment]
can we close this ?

File: pkgs/applications/editors/neovim/tests.nix [hunk 3]
```diff
@@ -171,4 +171,5 @@ rec {
     export HOME=$TMPDIR
     ${nvimWithLuaPackages}/bin/nvim -i NONE --noplugin -es
   '';
+
```
Comment (SuperSandro2000) [inline pkgs/applications/editors/neovim/tests.nix:174]: While we are at it. This should also be reverted.

**Comment by SuperSandro2000** [comment]
Closing on request of code owner.

**Pull request closed without merging**